omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,240 @@
+/* * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.idgen.web.controller;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.idgen.IdentifierPool;
+import org.openmrs.module.idgen.RemoteIdentifierSource;
+import org.openmrs.module.idgen.SequentialIdentifierGenerator;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+import static org.mockito.Matchers.endsWith;
+
+public class IdentifierSourceRestControllerTest extends MainResourceControllerTest {
+	
+    public static final String SEQUENTIAL_IDENTIFIER_SOURCE_UUID = "0d47284f-9e9b-4a81-a88b-8bb42bc0a901";
+    public static final String REMOTE_IDENTIFIER_SOURCE_UUID = "0d47284f-9e9b-4a81-a88b-8bb42bc0a902";
+    public static final String POOL_SOURCE_UUID = "0d47284f-9e9b-4a81-a88b-8bb42bc0a903";
+    public static final String IDENTIFIER_TYPE_UUID = "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c";
+    public static final String SEQUENTIAL_IDENTIFIER_SOURCE_TYPE = "SequentialIdentifierGenerator";
+    public static final String REMOTE_IDENTIFIER_SOURCE_TYPE = "RemoteIdentifierSource";
+    public static final String IDENTIFIER_POOL_SOURCE_TYPE = "IdentifierPool";
+
+    private IdentifierSourceService service;
```

Comment:
If you are using this in only one method, then there is no need of having it as a class level variable.

Reply:
Im using the service in multiple methods:

[getAllCount()](https://github.com/openmrs/openmrs-module-idgen/pull/37/files#diff-13128b20c5113cca7e0908cfb6ad9a13R68)
[shouldEditASequentialIdentifierSource()](https://github.com/openmrs/openmrs-module-idgen/pull/37/files#diff-13128b20c5113cca7e0908cfb6ad9a13R87)
[shouldEditARemoteIdentifierSource()](https://github.com/openmrs/openmrs-module-idgen/pull/37/files#diff-13128b20c5113cca7e0908cfb6ad9a13R95)
[shouldEditAnIdentifierPoolSource()](https://github.com/openmrs/openmrs-module-idgen/pull/37/files#diff-13128b20c5113cca7e0908cfb6ad9a13R103)